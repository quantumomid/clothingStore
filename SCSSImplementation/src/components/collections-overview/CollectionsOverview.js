import React from "react"
import CollectionPreview from "../../components/collection-preview/CollectionPreview"
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectShopCollectionsForPreview } from "../../redux/shop/shopSelector";
import "./CollectionsOverview.scss";

const CollectionsOverview = ({ collections }) => {
    console.log(collections);
    const collectionPreviewItems = collections.map(({ id, ...otherCollectionProps }) => <CollectionPreview key={id} {...otherCollectionProps} />)

    return(
        <div className="collections-overview">
            {collectionPreviewItems}
        </div>
    )
};

const mapStateToProps = createStructuredSelector({
    collections: selectShopCollectionsForPreview
});

export default connect(mapStateToProps)(CollectionsOverview);