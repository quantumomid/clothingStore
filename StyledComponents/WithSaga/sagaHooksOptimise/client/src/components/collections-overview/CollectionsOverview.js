import React from "react"
import CollectionPreview from "../../components/collection-preview/CollectionPreview"
import { useSelector } from "react-redux";
import { selectShopCollectionsForPreview } from "../../redux/shop/shopSelector";
import { CollectionsOverviewContainer } from "./collectionOverviewStyles";

const CollectionsOverview = () => {
    const collections = useSelector(selectShopCollectionsForPreview);
    // console.log(collections);
    const collectionPreviewItems = collections.map(({ id, ...otherCollectionProps }) => <CollectionPreview key={id} {...otherCollectionProps} />)

    return(
        <CollectionsOverviewContainer>
            {collectionPreviewItems}
        </CollectionsOverviewContainer>
    )
};

export default CollectionsOverview;