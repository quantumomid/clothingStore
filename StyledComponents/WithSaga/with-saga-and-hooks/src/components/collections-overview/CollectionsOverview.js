import React from "react"
import CollectionPreview from "../../components/collection-preview/CollectionPreview"
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectShopCollectionsForPreview } from "../../redux/shop/shopSelector";
import { CollectionsOverviewContainer } from "./collectionOverviewStyles";

const CollectionsOverview = ({ collections }) => {
    // console.log(collections);
    const collectionPreviewItems = collections.map(({ id, ...otherCollectionProps }) => <CollectionPreview key={id} {...otherCollectionProps} />)

    return(
        <CollectionsOverviewContainer>
            {collectionPreviewItems}
        </CollectionsOverviewContainer>
    )
};

const mapStateToProps = createStructuredSelector({
    collections: selectShopCollectionsForPreview
});

export default connect(mapStateToProps)(CollectionsOverview);