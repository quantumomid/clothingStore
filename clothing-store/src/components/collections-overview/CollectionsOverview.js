import React from "react"
import CollectionPreview from "../../components/collection-preview/CollectionPreview"
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectShopCollections } from "../../redux/shop/shopSelector";
import "./CollectionsOverview.scss";

const CollectionsOverview = ({ collections }) => {

    const collectionPreviewItems = collections.map(({ id, ...otherCollectionProps }) => <CollectionPreview key={id} {...otherCollectionProps} />)

    return(
        <div className="collections-overview">
            {collectionPreviewItems}
        </div>
    )
};

const mapStateToProps = createStructuredSelector({
    collections: selectShopCollections
});

export default connect(mapStateToProps)(CollectionsOverview);