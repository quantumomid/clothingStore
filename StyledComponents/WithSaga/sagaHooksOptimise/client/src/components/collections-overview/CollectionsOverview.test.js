import React from "react";
import { shallow } from "enzyme";
import CollectionsOverview from "./CollectionsOverview";
import * as redux from 'react-redux';

const mockCollections = [
    {
    id: 1,
    title: 'Hats',
    routeName: 'hats',
    items: [
      {
        id: 1,
        name: 'Brown Brim',
        imageUrl: 'https://i.ibb.co/ZYW3VTp/brown-brim.png',
        price: 25
      },
      {
        id: 2,
        name: 'Blue Beanie',
        imageUrl: 'https://i.ibb.co/ypkgK0X/blue-beanie.png',
        price: 18
      }
    ]
    },
    {
        id: 2,
        title: 'Sneakers',
        routeName: 'sneakers',
        items: [
          {
            id: 10,
            name: 'Adidas NMD',
            imageUrl: 'https://i.ibb.co/0s3pdnc/adidas-nmd.png',
            price: 220
          },
          {
            id: 11,
            name: 'Adidas Yeezy',
            imageUrl: 'https://i.ibb.co/dJbG1cT/yeezy.png',
            price: 280
          },
          {
            id: 12,
            name: 'Black Converse',
            imageUrl: 'https://i.ibb.co/bPmVXyP/black-converse.png',
            price: 110
          }
        ]
      }
]

describe("Testing the Collections Overview Component", () => {
    let wrapper;
    let spyOnUseSelector;

    beforeEach(() => {
        // Mock useSelector hook
        spyOnUseSelector = jest.spyOn(redux, 'useSelector');
        spyOnUseSelector.mockReturnValue(mockCollections);

        wrapper = shallow(<CollectionsOverview />)
    })

    afterEach(() => {
        jest.restoreAllMocks();
    });

    it("CollectionOverview Component should render", () => {
        expect(wrapper).toMatchSnapshot();
    })

})