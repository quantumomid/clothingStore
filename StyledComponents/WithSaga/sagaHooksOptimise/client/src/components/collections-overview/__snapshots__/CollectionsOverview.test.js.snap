// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Collections Overview Component CollectionOverview Component should render 1`] = `ShallowWrapper {}`;
